File: services/api/app/services/code_parser/model_types.py
from dataclasses import dataclass
from typing import Dict, List, Optional


@dataclass
class LayerNode:
    id: str
    name: str
    type: str
    params: Dict[str, str | int | float]
    input_shape: Optional[str] = None
    output_shape: Optional[str] = None
    activation: Optional[str] = None
    source_line: Optional[int] = None


@dataclass
class ParsedModel:
    layers: List[LayerNode]
    errors: List[str]

File: apps/web/src/lib/code-parser/model-types.ts
export type LayerType =
  | 'Input'
  | 'Linear'
  | 'Conv2d'
  | 'MaxPool2d'
  | 'AvgPool2d'
  | 'LSTM'
  | 'Dropout'
  | 'Flatten'
  | 'BatchNorm'
  | 'ReLU'
  | 'Sigmoid'
  | 'Softmax'
  | 'Tanh'
  | 'GELU'
  | 'LeakyReLU'
  | 'Unknown';

export type LayerNode = {
  id: string;
  name: string;
  type: LayerType;
  params: Record<string, string | number>;
  inputShape?: string;
  outputShape?: string;
  activation?: string;
  sourceLine?: number;
};

export type ParsedModel = {
  layers: LayerNode[];
  errors: string[];
};
